projects/dotcms-field-elements/src/utils/errorHandling.tsx

```diff
@@ -0,0 +1,56 @@
+// export class DotFieldParamError extends Error {
+//     constructor(propInfo: PropValidationInfo, expectedType: string) {
+//         super(
+//             `Warning: Invalid prop "${propInfo.name}" of type "${typeof propInfo.value}" supplied to "${propInfo.fieldType}" with the name "${propInfo.fieldName}", expected "${expectedType}"`
+//         );
+//         if (Error.captureStackTrace) {
+//             Error.captureStackTrace(this, DotFieldParamError);
+//         }
+//     }
+// }
+
+const fieldValidationMap = {
+    disabled: booleanValidator,
+    required: booleanValidator,
+    hint: stringValidator,
+    label: stringValidator,
+    name: stringValidator,
+    requiredMessage: stringValidator,
+    validationMessage: stringValidator
+};
+
+export interface PropValidationInfo {
```

Comment:
```suggestion
export interface PropValidationInfo<T> {
    field: {
        type: string;
        name: string;
    },
    prop: {
        name: string;
        value: <T>;
    }
}
```

I think we can do something like this and also use generics.